iroh-sync/src/store/fs.rs

```diff
@@ -59,10 +64,22 @@ const NAMESPACES_TABLE: TableDefinition<&[u8; 32], &[u8; 32]> =
 const RECORDS_TABLE: TableDefinition<RecordsId, RecordsValue> = TableDefinition::new("records-1");
 
 type RecordsId<'a> = (&'a [u8; 32], &'a [u8; 32], &'a [u8]);
+type RecordsIdOwned = ([u8; 32], [u8; 32], Bytes);
+
 type RecordsValue<'a> = (u64, &'a [u8; 64], &'a [u8; 64], u64, &'a [u8; 32]);
+
 type RecordsRange<'a> = TableRange<'a, RecordsId<'static>, RecordsValue<'static>>;
 type RecordsTable<'a> = ReadOnlyTable<'a, RecordsId<'static>, RecordsValue<'static>>;
-type DbResult<T> = Result<T, StorageError>;
+type RecordsReader<'a> = TableRangeReader<'a, RecordsId<'static>, RecordsValue<'static>>;
+
+// Records by key
+// Key: (NamespaceId, Key, AuthorId)
+// Value: same as in RECORDS_TABLE
```

Comment:
should we be duplicating the full value here? seems like we could should store a reference to the records table (not sure how to and if that is actually better..)

Reply:
Yeah, not sure myself. It's one extra lookup vs. 156 bytes (value size) per entry.

In the most recent commit, I changed it to not duplicate the value but instead do the extra lookup.